### PR TITLE
KAS-1399: Refactor Authentication Mixin

### PR DESCRIPTION
**Pull request by rafael-mindreau**
# ⚙️ Mixins Refactor: Authentication mixin

⚠️  **Word of warning** - This changes quite a lot of files, and has a potentially great impact on the functionality overall. In this introduction I attempt to share some insights into what could go wrong.

⛔Couldn't run tests locally...

## What has changed?

The `is-authenticated-mixin` has been phased out. It was used in 51 files. Thus this change has been very impactful.

### 🗒️ Overview of the mixin

The mixin contained the following key features:
image

What it does, is basically expose features from the `current-session` service with easy-to-use aliases. However, if you're new to the project, and haven't seen this before, you might easily go and wonder where the hell these come from if you see them popping up in templates. Another thing I found out on the way, is that the places that include this mixin **almost never** fully use all aliases. Most of the time only one alias. In some cases there was even no use at all, due to organic growth of the code and project.

### 🎛️ Method used

Mainly search and replace. I went and checked all usages per file. Component/Template pairs. Substitution of the alias prepended with the service that I would inject locally, and then unlink the mixin to see if it still worked. From the testing I've done locally, this all seemed to still work properly, but I can't stress enough that this change is quite critical.

### 🔧 Note on refactors

I've also taken the time to do some refactoring to Octane, and angle brackets in the templates where I could. This was done in a best-effort fashion, meaning I did this where I could. There were some places where this was tougher, or where computed properties were async. I've attempted to change those, but it introduced a lot of hassle, so much that I would just drop refactoring when encountering these.

The basic rule was:
1. Is there still another mixin on the component?
2. Are there async computed properties?
3. Other advanced constructions or observers?

Any of these would prevent me from going too deep. Later I just kept myself to the basic refactor of the mixin only, to save on time.

## 👀 What to look out for

* Any place requiring operations on the user role
* Any place requiring operations on the user object
* Any place checking if the current session is an Editor, Admin, or Viewer

### CODE REVIEW

File: app/components/agenda/agenda-header/template.hbs [hunk 1]
```diff
@@ -66,10 +70,14 @@
               showOn="click"
               animation="shift"
               placement="bottom"
-            as |attacher|
-            }}
+            as |attacher|}}
```
Comment (BenjaminDendas) [inline app/components/agenda/agenda-header/template.hbs]: Linter issue @ line 73: 
`  73:25  error  Incorrect indentation of close curly braces '}}' for the component '{{attach-popover}}' beginning at L73:C25. Expected '{{attach-popover}}' to be at L74:C12.  attribute-indentation`

**Review by BenjaminDendas — CHANGES_REQUESTED** [review]
Zeer lijvige PR met heel wat changes is, lijkt me normaal dat hier nog wat aan gedaan dient te worden.. Ik ben al scannend door de code gegaan omwille van de grootte en heb eerder zitten kijken of in de applicatie alles nog werkt zoals verwacht. Mijn bevindingen:

## 🐞 Bugs
### Switchen van agendaitem
Wanneer je switched van agendaitem dan wordt er een error geworpen in de console: 
image
Update: Dit blijkt een issue te zijn dat momenteel ook al op development zit.. dient wel aangepakt te worden.. Ik zal hier een ticket voor maken.

### Uploaden van documenten
Het uploaden van documenten werkt niet meer.
Agenda -> Agendaitem -> Documenten -> Upload document
image

### linken van reeds bezorgde documenten
Het linken van reeds bestaande documenten lukt niet meer:
image

### Een opmerking plaatsen
Een opmerking plaatsen geeft een error in de console:
image

### Authenticatie issues
Refactor zorgt ervoor dat je in een non-logged in state kan geraken waardoor je alles kan zien in kaleidos.
De sessie zit wel nog in de localstorage onder de key `ember-simple-auth-session` 
```{"authenticated":{"authenticator":"authenticator:mock-login","links":{"self":"/mock/sessions/current"},"data":{"type":"sessions","id":"5ea6dc5aa61bec000d000022","attributes":{"roles":["admin"]}},"relationships":{"account":{"links":{"related":"/accounts/f445824e-69d6-46ff-8803-718137ccbbad"},"data":{"type":"accounts","id":"f445824e-69d6-46ff-8803-718137ccbbad"}},"group":{"links":{"related":"/account-groups/71c068e6-d2f0-43de-93ab-cd1e7156ac4b"},"data":{"type":"account-groups","id":"71c068e6-d2f0-43de-93ab-cd1e7156ac4b"}}}}}```

image

**Comment by woutersf** [comment]
@ValenberghsSven  en @rafael-mindreau  Kunnen jullie de merge conflicts samen ook bekijken?

**Review by BenjaminDendas — CHANGES_REQUESTED** [review]
# Note:
door de app gegaan en vervolgens al scannend door de code

# 🐞Gevonden bugs

### Verwijderen opmerking
Het verwijderen van een opmerking vanuit een agenda-item geeft een foutmelding in de console die niet voorkomt op de development omgeving.
image

### Het switchen naar de notulen tab werkt niet meer waardoor de app blijft hangen.
Je krijgt de volgende error in de console en de applicatie reageerd hierna niet meer wanneer je switched naar eender welke tab
image

Soms krijg je ook deze (ik denk dat deze is omdat de app blijft hangen):
image

### Documenten wijzigen werkt niet meer
Dit dient nieuwe panel te tonen met de opties die je kan aanpassen..
**Branch:**
image
**Op Dev:(na het klikken op wijzigen) **
image

### Het downloaden van documentversies werkt niet meer
Je wordt redirect naar de niet gevonden pagina ..

image

### Documentversie downloaden vanaf het tabblad documenten van de agenda werkt niet (lijkt me hetzelfde component als bovenstaande issue
image

### Het wijzigen van de nota vanuit een agendaitem werkt niet meer
image
**Error:**
image

**Review by woutersf — APPROVED** [review]

**Comment by woutersf** [comment]
Voor mij is hij goed.

**Pull request merged**